citrination_client/views/client.py

```diff
@@ -12,3 +14,139 @@ def __init__(self, api_key, webserver_host="https://citrination.com", suppress_w
 
     def create(self, ml_template, search_template):
         pass
+
+    def get_available_columns(self, dataset_ids):
+        """
+        Retrieves the set of columns from the combination of dataset ids given
+
+        :param dataset_ids: The id of the dataset to retrieve columns from
+        :type dataset_ids: list of int
+        :return: A list of column names from the dataset ids given.
+        :rtype: list of str
+        """
+
+        data = {
+            "dataset_ids":
+                dataset_ids
+        }
+
+        failure_message = "Failed to get available columns in dataset(s) {}".format(dataset_ids)
+
+        return self._get_success_json(self._post_json(
+            'datasets/get-available-columns', data, failure_message=failure_message))['data']
+
+    def generate_search_template(self, dataset_ids):
+        """
+        Generates a default search templates from the available columns in the dataset ids given.
+
+        :param dataset_ids: The id of the dataset to retrieve files from
+        :type dataset_ids: list of int
+        :return: A search template based on the columns in the datasets given
+        """
+
+        data = {
+            "dataset_ids":
+                dataset_ids
+        }
+
+        failure_message = "Failed to generate a search template from columns in dataset(s) {}".format(dataset_ids)
+
+        return self._get_success_json(self._post_json(
+            'search_templates/from-dataset-ids', data, failure_message=failure_message))['data']
+
+    def prune_search_template(self, extract_as_keys, search_template):
```

Comment:
Same as above - I think we can hide this from users. `__prune_search_template`?

Reply:
Hide those, and expose a method that invokes those internally?